src/datasources/data-frame/datasources/v2/DataFrameDataSourceV2.ts

```diff
@@ -227,17 +227,24 @@ export class DataFrameDataSourceV2 extends DataFrameDataSourceBase {
         projection?: DataTableProjections[],
         substitutions?: string[]
     ): Observable<TableProperties[]> {
+        const projectionArray = projection || [];
```

Comment:
```suggestion
        const updatedProjections = projection;
```
1. Rename `projectionArray` as `updatedProjections`.
2. Assign `projection` to `[]` by default in the arguments.